src/stan-users-guide/latent-discrete.Rmd

```diff
@@ -1019,3 +1019,182 @@ the probability mass function of $z_i$ in the posterior.   These may
+function. For example, we may be interested in $\textsf{E}(X)$, the posterior 
+mean of the continuous parameters, where we would take $g(x, y) = x$. In this
+context, by *estimating*, we mean estimating in the *Monte Carlo* sense, which
+refers to calculating a series of low-dimensional summary statistics of the
+posterior distribution of the quantity of interest by processing the numerical
+values from a large number of posterior draws. This is in contrast to the usual
+frequentist approach to estimation that is concerned with bias and sampling
+variation of estimators that generate point estimates. When doing Bayesian 
+analyses, the Monte Carlo estimation procedure is often trivial to perform, 
+but more care is required when working with marginalized posteriors as we 
+describe in this section.
+
+If both $X$ and $Y$ are continuous, we would use Stan to sample $M$ draws
+from the joint posterior $p_{X, Y}(x, y)$ and then estimate $\mu$ with
+
+$$
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M {g(x_i, y_i)}.
+$$
+
+If $Y$ is discrete, however, we cannot use Stan to sample from the joint 
+posterior because Stan cannot compute the derivatives it needs for the 
+discrete parameters. 
+
+Instead, as outlined in the previous sections, we first marginalize out $Y$ 
+from the joint posterior to give the marginalized posterior $p_X(x)$. This
+marginalized posterior can then be implemented in Stan as usual, and Stan will 
+give samples $\{x_i\}_{i = 1}^M$ from the marginalized posterior. 
+
+How, using only the samples $\{x_i\}_{i = 1}^M$, can we estimate 
+$\textsf{E}(g(X, Y))$? We can use a conditional estimator. That is, for each
+function $g$ of interest we can compute
+
+$$
+h(X) = \textsf{E}(g(X, Y) \mid X)
+$$
+
+and then estimate $\textsf{E}(g(X, Y))$ with
+
+$$
+\hat{\mu} = \frac{1}{M} \sum_{i = 1}^M h(x_i).
+$$
+
+This estimator is justified by the fact that
+
+$$
+\textsf{E}(h(X)) 
+= \textsf{E}(\textsf{E}(g(X, Y) \mid X)) 
+= \textsf{E}(g(X, Y)) = \mu,
+$$
+
+which uses the [law of total expectation](https://en.wikipedia.org/wiki/Law_of_total_expectation). Using 
+this marginalized estimator gives us a way to estimate the expectation of any
+function $g$ for any combinations of discrete or continuous parameters in the 
+model. However, it presents a new challenge: evaluating the conditional 
+expectation $\textsf{E}(g(X, Y) \mid X)$.
+
+### Recovering the continuous parameters {-}
+
+In general, evaluating the expectation can be difficult, especially if we are
+interested in functions of both the discrete and continuous parameters. 
+Luckily, in practice we are commonly interested in expectations involving only 
+one of either the continuous or discrete parameters, for example the posterior
+mean of an individual scalar component of these parameters. In these cases,
+evaluating $\textsf{E}(g(X, Y) \mid X)$ becomes tractable.
+
+If $g(X, Y)$ is a function of only the continuous parameters $X$, the 
```

Comment:
Is it usefully illustrative or just cluttering and confusing to explain how this works for the continuous parameters?  This doesn't depend on the existence of discrete parameters at all.  It's just propagating uncertainty to functions of parameters in the usual way.

Reply:
I agree that this could be removed.  I just felt like I'd been so nitpick on everything else that I had to stop somewhere.

Reply:
To me, this explanation seems like helpful scaffolding for readers who are less familiar with the concepts described in this section.  This part here starts with something that should be familiar, to help ease readers into the less familiar parts that follow.